kie-internal/src/main/java/org/kie/internal/utils/ChainedProperties.java

```diff
@@ -121,25 +114,14 @@ public ChainedProperties(String confFileName,
                         this.props, confClassLoader );
 
         confClassLoader = Thread.currentThread().getContextClassLoader();
-        if ( confClassLoader != null && confClassLoader != classLoader ) {
-            loadProperties( getResources( "META-INF/drools." + confFileName,
-                                          confClassLoader ),
-                            this.props, confClassLoader );
-            loadProperties( getResources( "/META-INF/drools." + confFileName,
-                                          confClassLoader ),
-                            this.props, confClassLoader );
-        }
-        
-        confClassLoader = ClassLoader.getSystemClassLoader();
-        if ( confClassLoader != null && confClassLoader != classLoader ) {
-            loadProperties( getResources( "META-INF/drools." + confFileName,
-                                          confClassLoader ),
-                            this.props, confClassLoader );
-            loadProperties( getResources( "/META-INF/drools." + confFileName,
-                                          confClassLoader ),
-                            this.props, confClassLoader );
+        loadPropertiesFromClassLoader( confFileName, classLoader, confClassLoader );
+
+        try {
+            confClassLoader = ClassLoader.getSystemClassLoader();
+            loadPropertiesFromClassLoader( confFileName, classLoader, confClassLoader );
+        } catch (Throwable t) {
```

Comment:
I'd catch SecurityException.
RuntimeException would be fine too (although I think its' too much)
Throwable I don't think is fine as it eats errors, such as OutOfMemoryError (which you really want to fail fast)